Pipeline to generate the release and pass all the tests

miguelgfierro opened the pull request:
### Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->


### Related Issues
<!--- If it fixes an open issue, please link to the issue here. -->
1408

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [ ] This PR is being made to `staging branch` and not to `main branch`.

anargyri reviewed (commented):
Maybe we want to keep dev-requirements.txt ? Or change CONTRIBUTING.md where we reference it.

anargyri commented on tests/ci/azure_pipeline_test/release_pipeline.yml:
```diff
@@ -0,0 +1,89 @@
+    - unit
+    task_name: "Test - Unit Notebook Linux Spark"
+    conda_env: "relase_unit_notebook_linux_spark"
+    conda_opts: "python=3.6"
```
I think we can start using 3.7, is there any obstacle to that?

miguelgfierro replied:
yeah I can create another  set with 3.7, I'll try to create a matrix

miguelgfierro commented:
> Maybe we want to keep dev-requirements.txt ? Or change CONTRIBUTING.md where we reference it.

I removed this because it was only installing pytest and black, and we were only installing pytest, so it is added directly into the pipeline and we removed an extra file we don't really need

anargyri commented:
> > Maybe we want to keep dev-requirements.txt ? Or change CONTRIBUTING.md where we reference it.
> 
> I removed this because it was only installing pytest and black, and we were only installing pytest, so it is added directly into the pipeline and we removed an extra file we don't really need

Yes, but besides the pipeline, it can be used by people who contribute to the repo. This is why we reference it in CONTRIBUTING.md.
